soh/src/overlays/actors/ovl_En_Ex_Ruppy/z_en_ex_ruppy.c

```diff
@@ -379,19 +425,30 @@ void EnExRuppy_Draw(Actor* thisx, GlobalContext* globalCtx) {
     static void* rupeeTextures[] = {
         gRupeeGreenTex, gRupeeBlueTex, gRupeeRedTex, gRupeePinkTex, gRupeeOrangeTex,
     };
+    static void* rupeeTexturesNew[] = {
+        GID_RUPEE_GREEN, GID_RUPEE_BLUE, GID_RUPEE_RED, GID_RUPEE_PURPLE, GID_RUPEE_GOLD,
+    };
     s32 pad;
     EnExRuppy* this = (EnExRuppy*)thisx;
 
     if (!this->invisible) {
-        OPEN_DISPS(globalCtx->state.gfxCtx, "../z_en_ex_ruppy.c", 774);
+        OPEN_DISPS(globalCtx->state.gfxCtx, __FILE__, __LINE__);
 
         func_80093D18(globalCtx->state.gfxCtx);
         func_8002EBCC(thisx, globalCtx, 0);
-        gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_en_ex_ruppy.c", 780),
-                  G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
-        gSPSegment(POLY_OPA_DISP++, 0x08, SEGMENTED_TO_VIRTUAL(rupeeTextures[this->colorIdx]));
-        gSPDisplayList(POLY_OPA_DISP++, gRupeeDL);
+        gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx, __FILE__, __LINE__), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+        if (CVar_GetS32("gNewDrops", 0) !=0) {
+            if (this->type == 4 && this->colorIdx >= 3) {
+                //For some reason the red rupee target become purple.
+                //when using new drops it will show as Gold and that wrong it need to be red.
+                this->colorIdx = 2;
+            }
+            GetItem_Draw(globalCtx, rupeeTexturesNew[this->colorIdx]);
+        } else {
+            gSPSegment(POLY_OPA_DISP++, 0x08, SEGMENTED_TO_VIRTUAL(rupeeTextures[this->colorIdx]));
+            gSPDisplayList(POLY_OPA_DISP++, gRupeeDL);
+        }
 
-        CLOSE_DISPS(globalCtx->state.gfxCtx, "../z_en_ex_ruppy.c", 784);
+        CLOSE_DISPS(globalCtx->state.gfxCtx, __FILE__, __LINE__);
```

Comment:
As discussed, I think we should only make these changes in batch on the whole codebase.

Reply:
since I modify line count shouldn't it be present ? 
or having wrong line number is not a problem ?

Reply:
I think the point is that it's best the project stick to one convention, even if it's a less ideal convention, than be inconsistent. My guess at what would be the course of action in this case is either A. update the line count instead of using __LINE__ or B. make another PR that makes the __FILE__, __LINE__ change to every file that uses CLOSE_DISPS, wait for that PR to be merged, and then fix any conflicts that may then exist between this PR and develop.

Reply:
#559 is the intended direction if that validated.
depending on result this PR will be updated to match the result :)